### ISSUE TITLE
Update payment flow from `settings/subscription` route to update trial subscription instead of creating a new one

### ISSUE THREAD

**Issue by p6l-richard** [agreeto/agreeto#18]
We're currently creating two subscriptions:
Thanks.

There actually is an issue about creating multiple subscriptions for the user -- which isn't exactly ideal.

[This StackOverflow user](https://stackoverflow.com/a/70071633/5608461) proposes to:

- create the session in `mode: setup` to collect payment details & nothing more,
- set up a resolver for the checkout.session.completed event that's coming in through our webhook, and
- [update the existing subscription](https://stripe.com/docs/api/subscriptions/update) by calling something like:
```
await stripe.subscriptions.update(
  'sub_1M9YqBEdYMdxQoGeTqqP4NiA',
  {metadata: {...}}
);
```
 ![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/17e55763-3566-4aa1-a5f0-0aaeeee794fc/Untitled.png)
 
 
 Basically to simulate the following (apparently not possible to set  `mode: upgrade`):
https://cdn.discordapp.com/attachments/1025507285553922078/1047260536011755583/CleanShot_2022-11-29_at_22.18.18.png
